core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -651,60 +651,126 @@ class UserService(protected val ctx: RawlsRequestContext,
       }
     }
 
+  private def getLegacyBillingPolicies(samRole: ProjectRole): Seq[SamResourcePolicyName] =
+    samRole match {
+      case ProjectRoles.Owner => Seq(SamBillingProjectPolicyNames.owner)
+      case ProjectRoles.User =>
+        Seq(SamBillingProjectPolicyNames.workspaceCreator, SamBillingProjectPolicyNames.canComputeUser)
+    }
+  private def getV2BillingPolicy(samRole: ProjectRole): SamResourcePolicyName =
+    samRole match {
+      case ProjectRoles.Owner => SamBillingProjectPolicyNames.owner
+      case ProjectRoles.User  => SamBillingProjectPolicyNames.workspaceCreator
+    }
+
   def addUserToBillingProject(projectName: RawlsBillingProjectName,
                               projectAccessUpdate: ProjectAccessUpdate
   ): Future[Unit] =
     requireProjectAction(projectName, SamBillingProjectActions.alterPolicies) {
-      val policies = projectAccessUpdate.role match {
-        case ProjectRoles.Owner => Seq(SamBillingProjectPolicyNames.owner)
-        case ProjectRoles.User =>
-          Seq(SamBillingProjectPolicyNames.workspaceCreator, SamBillingProjectPolicyNames.canComputeUser)
+      val policies = getLegacyBillingPolicies(projectAccessUpdate.role)
+      addUserToBillingProjectInner(projectName, projectAccessUpdate, policies)
+    }
+
+  private def addUserToBillingProjectInner(projectName: RawlsBillingProjectName,
+                                           projectAccessUpdate: ProjectAccessUpdate,
+                                           policies: Seq[SamResourcePolicyName]
+  ): Future[Unit] =
+    for {
+      _ <- Future.traverse(policies) { policy =>
+        samDAO
+          .addUserToPolicy(SamResourceTypeNames.billingProject,
+                           projectName.value,
+                           policy,
+                           projectAccessUpdate.email,
+                           ctx.userInfo
+          )
+          .recoverWith { case regrets: Throwable =>
+            if (policy == SamBillingProjectPolicyNames.canComputeUser) {
+              logger.info(
+                s"error adding user to canComputeUser policy for $projectName likely because it is a v2 billing project which does not have a canComputeUser policy. regrets: ${regrets.getMessage}"
+              )
+              Future.successful(())
+            } else {
+              Future.failed(regrets)
+            }
+          }
       }
+    } yield {}
 
+  def addUserToBillingProjectV2(projectName: RawlsBillingProjectName,
```

Comment:
Does orchestration need to be updated to point to these new method? The orchestration Swagger already says that the `operationID` for the v2 version is `addUserToBillingProjectV2` (which didn't previously exist). I don't understand though how the dispatching of calls happens-- do we need to change orchestration to actually dispatch to these new methods?

Reply:
The operationId refers to [this](https://github.com/broadinstitute/rawls/blob/8552f25e1bcf0b68c0d77ede2d3758609717a382/core/src/main/resources/swagger/api-docs.yaml#L522-L586) endpoint in the _rawls_ swagger. We should verify if firecloud-orch is indeed calling this endpoint.

Reply:
Oh right, orchestration calls the Rawls REST API.

This is what orch is calling. I don't see that we call the v2 version of _anything_, and I don't see that there are pass throughs for much of what is in the Swagger related to billing. Is the Swagger just all bogus (@aherbst-broad , @marctalbott )?

```
  private def editBillingMembershipURL(projectId: String, role: ProjectRole, email: String) = {
    FireCloudConfig.Rawls.authUrl + s"/billing/$projectId/${role.toString}/${java.net.URLEncoder.encode(email, "UTF-8")}"
  }
```

Reply:
(Will discuss at tech talk)